jwst/lib/set_telescope_pointing.py

```diff
@@ -148,6 +144,77 @@ def update_wcs(model, default_pa_v3=0., siaf_path=None, **kwargs):
         Keyword arguments used by matrix calculation routines.
     """
 
+    # If the type of exposure is not FGS, then attempt to get pointing
+    # from telemetry.
+    if model.meta.exposure.type.lower().startswith('fgs'):
```

Comment:
Can't just use a simple test based on `fgs` being the first 3 characters, because science mode exposures from the FGS also start with those characters (e.g. FGS_IMAGE). So you need a specific list of guide-mode EXP_TYPE's, which is FGS_ID-IMAGE, FGS_ID-STACK, FGS_ACQ1, FGS_ACQ2, FGS_TRACK, and FGS_FINEGUIDE.